gobblin-core/src/main/java/gobblin/writer/http/AsyncHttpWriterBuilder.java

```diff
@@ -44,11 +52,12 @@
  * @param <RQ> type of request
  * @param <RP> type of response
  */
+@Slf4j
 public abstract class AsyncHttpWriterBuilder<D, RQ, RP> extends FluentDataWriterBuilder<Void, D, AsyncHttpWriterBuilder<D, RQ, RP>> {
   public static final String CONF_PREFIX = "gobblin.writer.http.";
 
   @Getter
-  protected State state;
+  protected WorkUnitState state;
```

Comment:
I was considering the builder may accept other kinds of `State`, not coupled with `WorkUnitState`. In fact, it aligns with the whole class family from `DataWriterBuilder`, which provides `State`.  In this scenario, the builder is given a type of `State`. I don't think this is a wrong abstraction.

You're right. It's not creating a broker but `getBroker()`